### PR TITLE
Fix/badge find

### PR DESCRIPTION
**Pull request by joeltrew**
Updates BadgeController find method + adds generics and nullability

### CODE REVIEW

File: ThunderCloud/TSCBadgeController.h [hunk 1]
```diff
@@ -21,41 +22,41 @@
...
  */
-- (BOOL)hasEarntBadgeWithId:(NSString *)badgeId;
+- (BOOL)hasEarntBadgeWithId:(nonnull NSString *)badgeId;
 
 /**
  Marks and saves a badge as earnt
  @param badgeId The unique id for the `TSCBadge`
  */
-- (void)markBadgeAsEarnt:(NSString *)badgeId;
+- (void)markBadgeAsEarnt:(nullable NSString *)badgeId;
```
Comment (MattCheetham) [inline ThunderCloud/TSCBadgeController.h]: Shouldn't this be non-null? Is there a benefit of letting null be passed to mark a badge as earnt?

Comment (joeltrew) [inline ThunderCloud/TSCBadgeController.h]: `- (void)markBadgeAsEarnt:(NSString *)badgeId
{
    if (badgeId && ![self hasEarntBadgeWithId:badgeId]) {`

We check if the badgeId is nil in the method body so i assumed there are some places where nil is passed in?

Comment (joeltrew) [inline ThunderCloud/TSCBadgeController.h]: https://github.com/3sidedcube/iOS-ThunderCloud/blob/develop/ThunderCloud/TSCBadgeController.m#L105

Comment (MattCheetham) [inline ThunderCloud/TSCBadgeController.h]: Yeah I guess it's a bad idea to nonnull it at this point and end up in trouble for passing in nil later on :/

**Pull request merged**